astropy/units/structured.py

```diff
@@ -239,8 +240,10 @@ def _recursively_apply(self, func, cls=None):
             will return a new `~astropy.units.StructuredUnit` instance.
         """
         applied = tuple(func(part) for part in self.values())
-        # Once not NUMPY_LT_1_23: results = np.void(applied, self._units.dtype).
-        results = np.array(applied, self._units.dtype)[()]
+        if sys.version_info < (3, 10):
```

Comment:
Interesting that this is python-version dependent!  I assume you found out the hard way that python 3.9 does not work?

Reply:
Yeah! Only the Python 3.9 test failed, complaining about void no5 accepting 2 arguments.

Reply:
Hm. I rebased the py3.10 bump PR and found it's failing on 3.10. https://github.com/astropy/astropy/pull/15603.
So actually the `NUMPY_LT_1_23` label might be incorrect.

Reply:
According to https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.void, the second argument was added in numpy 1.24, so I guess the correct guard is `if NUMPY_LT_1_24`

Reply:
Fixed!

Reply:
And folded it into https://github.com/astropy/astropy/pull/15706/commits/4e49989c19fc3287b7db57edb933a1346ec3f651

Reply:
OK, sorry that my comment there was so misleading!

Reply:
It was probably planned to be done in numpy 1.23 but bumped to numpy 1.24.!